### PR TITLE
Add a few more sanity tests

### PR DESCRIPTION
**Pull request by atodorov**
--- Description of proposed changes ---

A few more sanity tests inspired by official documentation.

--- Merge policy ---

- [ ] Travis CI PASS
- [ ] `*-aws-runtest` PASS
- [ ] `*-azure-runtest` PASS
- [ ] `*-images-runtest` PASS
- [ ] `*-openstack-runtest` PASS
- [ ] `*-vmware-runtest` PASS
- [ ] For `rhel8-*` and `rhel7-*` branches commit log references an approved
  bug in Bugzilla. Do not merge if the bug doesn't have the 3 ACKs set to `+`!

--- Jenkins commands ---

- `ok to test` to accept this pull request for testing
- `test this please` for a one time test run
- `retest this please` to start a new build

### CODE REVIEW

File: tests/cli/test_compose_sanity.sh [hunk 1]
```diff
@@ -36,6 +36,11 @@ rlJournalStart
 
             rlRun -t -c "$CLI compose image $UUID"
             rlAssertExists "$UUID-root.tar.xz"
+
+            # because this path is listed in the documentation
+            rlAssertExists    "/var/lib/lorax/composer/results/$UUID/"
+            rlAssertExists    "/var/lib/lorax/composer/results/$UUID/$UUID-root.tar.xz"
+            rlAssertNotDiffer "/var/lib/lorax/composer/results/$UUID/$UUID-root.tar.xz" "$UUID-root.tar.xz"
```
Comment (bcl) [inline tests/cli/test_compose_sanity.sh]: Inside results/UUID/ it is just root.tar.xz
The file is only named $UUID-root.tar.xz when retrieved with composer-cli compose image $UUID

**Comment by atodorov** [comment]
@bcl I've changed `[[packages]]` to `[[modules]]` and now the auto-update functionality seems to work. Question is what is the difference between the two and why adding packages to the blueprint causes push to fail ?

**Comment by bcl** [comment]
It shouldn't matter. And actually we should really be using [[packages]] for everything since we really don't have module support yet, and the format of that will change at some point (although we will maintain backward compatibility). Do you have the logs from the failed test, I didn't see them.

**Comment by atodorov** [comment]
@bcl here is the log only for the blueprints sanity test and nothing else:

```
# cat server.log 
- - - [2019-04-11 03:49:29] "GET /api/status HTTP/1.1" 200 - 0.002311
- - - [2019-04-11 03:49:29] "GET /api/status HTTP/1.1" 200 - 0.000698
- - - [2019-04-11 03:49:29] "GET /api/status HTTP/1.1" 200 - 0.000687
- - - [2019-04-11 03:49:29] "GET /api/status HTTP/1.1" 200 - 0.000701
- - - [2019-04-11 03:49:29] "GET /api/status HTTP/1.1" 200 - 0.000708
- - - [2019-04-11 03:49:29] "GET /api/status HTTP/1.1" 200 - 0.000732
- - - [2019-04-11 03:49:29] "GET /api/status HTTP/1.1" 200 347 0.000719
- - - [2019-04-11 03:49:30] "GET /api/v0/blueprints/list?limit=0 HTTP/1.1" 200 270 0.014673
- - - [2019-04-11 03:49:30] "GET /api/v0/blueprints/list?limit=3 HTTP/1.1" 200 330 0.001553
- - - [2019-04-11 03:49:30] "GET /api/v0/blueprints/list?limit=0 HTTP/1.1" 200 270 0.003194
- - - [2019-04-11 03:49:30] "GET /api/v0/blueprints/list?limit=3 HTTP/1.1" 200 330 0.001714
- - - [2019-04-11 03:49:30] "GET /api/v0/blueprints/list?limit=0 HTTP/1.1" 200 270 0.003204
- - - [2019-04-11 03:49:30] "GET /api/v0/blueprints/list?limit=3 HTTP/1.1" 200 330 0.001310
- - - [2019-04-11 03:49:30] "GET /api/v0/blueprints/info/example-http-server?format=toml HTTP/1.1" 200 745 0.018146
- - - [2019-04-11 03:49:31] "GET /api/v0/blueprints/info/non-existing-bp?format=toml HTTP/1.1" 400 368 0.005064
- - - [2019-04-11 03:49:32] "POST /api/v0/blueprints/new HTTP/1.1" 400 321 0.005335
- - - [2019-04-11 03:49:32] "GET /api/v0/blueprints/list?limit=0 HTTP/1.1" 200 270 0.003209
- - - [2019-04-11 03:49:32] "GET /api/v0/blueprints/list?limit=3 HTTP/1.1" 200 330 0.001517
- - - [2019-04-11 03:49:32] "GET /api/v0/blueprints/info/beakerlib?format=toml HTTP/1.1" 400 356 0.004613
- - - [2019-04-11 03:49:32] "GET /api/v0/blueprints/info/beakerlib?format=toml HTTP/1.1" 400 356 0.004907
- - - [2019-04-11 03:49:33] "POST /api/v0/blueprints/new HTTP/1.1" 400 322 0.005494
- - - [2019-04-11 03:49:33] "GET /api/v0/blueprints/info/beakerlib?format=toml HTTP/1.1" 400 356 0.003857
- - - [2019-04-11 03:49:33] "DELETE /api/v0/blueprints/delete/beakerlib HTTP/1.1" 400 370 0.003290
- - - [2019-04-11 03:49:33] "GET /api/v0/blueprints/list?limit=0 HTTP/1.1" 200 270 0.003245
- - - [2019-04-11 03:49:33] "GET /api/v0/blueprints/list?limit=3 HTTP/1.1" 200 330 0.001681


# cat composer.log
2019-04-11 03:47:43,865 DEBUG lorax-composer: opts=Namespace(BLUEPRINTS='/var/lib/lorax/composer/blueprints/', config='/etc/lorax/composer.conf', group='weldr', logfile='/var/log/lorax-composer/composer.log', mockfiles='/var/tmp/bdcs-mockfiles/', no_system_repos=False, proxy=None, releasever=None, sharedir=None, showver=False, socket='/run/weldr/api.socket', tmp='/var/tmp', user='weldr')
2019-04-11 03:47:43,866 INFO pylorax: selinux is enabled and in Enforcing mode
2019-04-11 03:47:43,889 DEBUG lorax-composer: user is now (990, 990, 990):(990, 990, 990)
2019-04-11 03:47:43,929 INFO lorax-composer: releasever = 29
2019-04-11 03:47:43,931 INFO lorax-composer: Using platform:f29 for module_platform_id
2019-04-11 03:47:43,960 INFO lorax-composer: Updating repository metadata
2019-04-11 03:49:29,525 INFO lorax-composer: Starting lorax-composer-29.27-1 on /run/weldr/api.socket with blueprints from /var/lib/lorax/composer/blueprints/
2019-04-11 03:49:31,829 ERROR lorax-composer: (v0_blueprints_info) No commits for non-existing-bp.toml on the master branch.
2019-04-11 03:49:32,149 ERROR lorax-composer: (v0_blueprints_new) <string>(9, 1): key_table_conflict
2019-04-11 03:49:32,648 ERROR lorax-composer: (v0_blueprints_info) No commits for beakerlib.toml on the master branch.
2019-04-11 03:49:32,862 ERROR lorax-composer: (v0_blueprints_info) No commits for beakerlib.toml on the master branch.
2019-04-11 03:49:33,041 ERROR lorax-composer: (v0_blueprints_new) <string>(13, 1): key_table_conflict
2019-04-11 03:49:33,307 ERROR lorax-composer: (v0_blueprints_info) No commits for beakerlib.toml on the master branch.
2019-04-11 03:49:33,533 ERROR lorax-composer: (v0_blueprints_delete) ggit-error: failed to remove entry: file isn't in the tree - beakerlib.toml (-1)
2019-04-11 03:49:35,265 DEBUG lorax-composer: opts=Namespace(BLUEPRINTS='/var/lib/lorax/composer/blueprints/', config='/etc/lorax/composer.conf', group='weldr', logfile='/var/log/lorax-composer/composer.log', mockfiles='/var/tmp/bdcs-mockfiles/', no_system_repos=False, proxy=None, releasever=None, sharedir=None, showver=False, socket='/run/weldr/api.socket', tmp='/var/tmp', user='weldr')
2019-04-11 03:49:35,265 INFO pylorax: selinux is enabled and in Enforcing mode
2019-04-11 03:49:35,276 DEBUG lorax-composer: user is now (990, 990, 990):(990, 990, 990)
2019-04-11 03:49:35,305 INFO lorax-composer: releasever = 29
2019-04-11 03:49:35,307 INFO lorax-composer: Using platform:f29 for module_platform_id
2019-04-11 03:49:35,350 INFO lorax-composer: Updating repository metadata
2019-04-11 03:49:52,912 INFO lorax-composer: Starting lorax-composer-29.27-1 on /run/weldr/api.socket with blueprints from /var/lib/lorax/composer/blueprints/
```

The relevant line I think is this:

```
2019-04-11 03:49:33,041 ERROR lorax-composer: (v0_blueprints_new) <string>(13, 1): key_table_conflict
```


And here's how `beakerlib.toml` looks like just before the test tries to push the blueprint for the first time:

```
name = "beakerlib"
description = "Start building tests with beakerlib."
version = "0.0.1"
packages = []
groups = []

[[packages]]
name = "beakerlib"
version = "*"
```

I believe the offending part is `packages = []` and later using `[[packages]]` so my next commit will change this.

**Comment by bcl** [comment]
> I believe the offending part is `packages = []` and later using `[[packages]]` so my next commit will change this.

Correct, you can't set it to be empty and have entries at the same time. Also, when creating the blueprint you don't need to set empty lists, so modules = [] and groups = [] isn't needed.

**Comment by atodorov** [comment]
Empty groups and modules are needed so that I can directly compare the file contents against `blueprints show`

**Comment by atodorov** [comment]
@bcl do you approve the latest changes so I can backport to downstream branches ?

**Pull request merged**